packages/mobx-state-tree/src/core/node/object-node.ts

```diff
@@ -238,9 +238,20 @@ export class ObjectNode<C, S, T> extends BaseNode<C, S, T> {
         this._childNodes = EMPTY_OBJECT
 
         this.state = NodeLifeCycle.CREATED
-        this.fireHook(Hook.afterCreate)
 
-        this.finalizeCreation()
+        if (runHooks) {
+            this.fireHook(Hook.afterCreate)
+            // Note that the parent might not be finalized at this point
+            // so afterAttach won't be called in that case
+            this.finalizeCreation()
+
+            // run the hooks of the parents that we created
+            for (const p of parentChain.reverse()) {
```

Comment:
This isn't very efficient. But neither is `parentChain.unshift(parent)` above. I'd assume the depth of the tree is small enough in all cases, that there is no need to optimize this.

Reply:
Will this possibly fire the parent afterCreates multiple times?

Reply:
I don't believe it can fire more than once because the use of the `_observableInstanceState`. The `parentChain` only includes the parents with an `_observableInstanceState` of `UNINITIALIZED`. See line 203 above. 

The current node's `_observableInstanceState` is set to `CREATING` immediately when `createObservableInstance` is called. So the current node will not be included in any `parentChain`s of its children.

Additionally parents in the `parentChain` have their observable instance created for the first time on line 212 (It was at this point in the old code that the `afterCreate` hook was called on the parents before the children). So these parents will now have an `_observableInstanceState` of `CREATING`. Therefore any future `parentChain`s will not include these objects and thus their `afterCreate` will not be called a second time.